purenix/src/Lib.hs

```diff
@@ -2,28 +2,35 @@ module Lib where
 
 import Data.Aeson (decode)
 import Data.Aeson.Types (parseEither)
+import qualified Data.Text as T
 import Data.Text.Lazy (pack)
 import Data.Text.Lazy.Encoding (encodeUtf8)
+import qualified Data.Text.Lazy.IO as TL
 import Language.PureScript.CoreFn.FromJSON (moduleFromJSON)
+import Nix.Convert (convert)
+import Nix.Print (renderExpr)
 import System.Directory (createDirectoryIfMissing)
-import Text.Pretty.Simple (pPrint)
+import qualified System.Environment as Env
+import qualified System.Exit as Sys
 
 defaultMain :: IO ()
 defaultMain = do
+  Env.getArgs >>= print
   corefn <- readFile "../purescript-cabal-parser/output/Main/corefn.json"
+  -- TODO why is this lazy?
   let maybeValue = decode $ encodeUtf8 $ pack corefn
```

Comment:
Maybe you're asking here why `pack` is `Data.Text.Lazy.pack` and not just `Data.Text.pack`?

`Data.Aeson.decode` takes a lazy `ByteString` as input, so I used `Data.Text.Lazy.pack`.

Although now when I pull up the aeson haddocks and scroll down the page a little more, I see there is a `decodeStrict` I could have used instead:

https://hackage.haskell.org/package/aeson-1.5.6.0/docs/Data-Aeson.html#v:decodeStrict

Also, I'm sure you can tell, but this file was basically just a quick hack to show how I imagined the `spago build` workflow to work.  Feel free to fix/change anything here.